ggml/src/ggml.c

```diff
@@ -6414,48 +6404,6 @@ static size_t gguf_type_size(enum gguf_type type) {
     return GGUF_TYPE_SIZE[type];
 }
 
-static bool gguf_tensor_info_sanitize(struct gguf_tensor_info * info) {
```

Comment:
Is there a comprehensive replacement for these checks? This is done to avoid security issues when loading ggufs, and it is of critical importance to keep them.

Reply:
The checks are all still there. I moved those that deal with tensor dimensions out of the dedicated function because bad values would lead to writes of arbitrary size to the stack (and `gguf_tensor_info_sanitize` was called only afterwards). And then there was basically nothing left in the function and it was only being called once so I moved the rest of the code too.